extensions/wikia/Search/WikiaSearchController.class.php

```diff
@@ -696,31 +631,50 @@ protected function setResponseValuesFromConfig( Wikia\Search\Config $searchConfi
 		} else {
 			$this->setVal( 'mediaData', [] );
 		}
-		if ( $this->wg->OnWikiSearchIncludesWikiMatch && $searchConfig->hasWikiMatch() ) {
+
+		if ( $searchConfig->hasWikiMatch() ) {
 			$this->registerWikiMatch( $searchConfig );
 		}
-		$topWikiArticlesHtml = '';
-
-		if ( ! $searchConfig->getInterWiki() && $wgLanguageCode == 'en'
-			&& !$isMonobook ) {
-			$dbname = $this->wg->DBName;
-			$cacheKey = wfMemcKey(
-				__CLASS__,
-				'WikiaSearch',
-				'topWikiArticles',
-				$this->wg->CityId,
-				static::TOP_ARTICLES_CACHE,
-				$isGridLayoutEnabled
-			);
-			$topWikiArticlesHtml = WikiaDataAccess::cache(
-				$cacheKey,
-				86400 * 5, // 5 days, one business week
-				function () {
-					return $this->app->renderView( 'WikiaSearchController', 'topWikiArticles' );
-				}
-			);
+
+		$this->addRightRailModules( $searchConfig );
+	}
+
+	protected function addRightRailModules( Wikia\Search\Config $searchConfig ) {
+		global $wgLang;
+
+		$isMonobook = $this->response->getVal( 'isMonobook' );
+		$query = $searchConfig->getQuery()->getSanitizedQuery();
+		$this->response->setValues( [
+			'fandomStories' => [],
+			'topWikiArticles' => [],
+		] );
+
+		if ( $searchConfig->getInterWiki() || $isMonobook ) {
+			return;
+		}
+
+		if ( $wgLang->getCode() === 'en' && !empty( $query ) ) {
+			$fandomStories = \Wikia\Search\FandomSearch::getStoriesWithCache( $query );
+			$viewMoreFandomStoriesLink = \Wikia\Search\FandomSearch::getViewMoreLink( $fandomStories, $query );
+
+			if ( !empty( $fandomStories ) ) {
```

Comment:
Additionally this check is handy for requests that don't render template - API calls.